app/src/main/java/com/cornellappdev/android/eatery/presenter/MainPresenter.java

```diff
@@ -0,0 +1,75 @@
+package com.cornellappdev.android.eatery.presenter;
+
+import android.util.Log;
+
+import com.cornellappdev.android.eatery.Repository;
+import com.cornellappdev.android.eatery.model.EateryBaseModel;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+
+public class MainPresenter{
+
+	public Repository rInstance = Repository.getInstance();
+
+	public void setEateryList(ArrayList<EateryBaseModel> eateryList) {
+		rInstance.setEateryList(eateryList);
+		rInstance.setSearchList(rInstance.getEateryList());
+	}
+
+	public String query = "";
+
+	private void searchList(String query) {
+		final String lowercaseQuery = query.toLowerCase();
+		for (EateryBaseModel model : rInstance.getSearchList()) {
+			final HashSet<String> mealSet = model.getMealItems();
+
+			boolean foundNickName = false;
+			if (model.getNickName().toLowerCase().contains(lowercaseQuery)) {
+				foundNickName = true;
+			}
+
+			ArrayList<String> matchedItems = new ArrayList<>();
+			boolean foundItem = false;
+			for (String item : mealSet) {
+				if (item.toLowerCase().contains(lowercaseQuery)) {
+					foundItem = true;
+					matchedItems.add(item);
+				}
+			}
+
+			if (model.matchesFilter() && (foundItem || foundNickName)) {
+				if (foundNickName) {
+					model.setSearchedItems(new ArrayList<>(mealSet));
+				} else {
+					model.setSearchedItems(matchedItems);
+				}
+				model.setMatchesSearch(true);
+			} else {
+				model.setMatchesSearch(false);
+			}
+		}
+	}
+
+    public void setQuery(String query){
```

Comment:
`(String query) {`, make sure that linter is configured!

Reply:
it is now!